src/args.rs

```diff
@@ -49,17 +64,48 @@ impl Args {
         self.paths.is_empty()
     }
 
-    pub(crate) fn output_mode(&self) -> OutputMode {
-        if self.vimgrep {
-            OutputMode::Vimgrep
+    fn line_number(&self) -> bool {
+        true
+    }
+
+    fn per_match(&self) -> bool {
+        self.vimgrep
+    }
+
+    fn per_match_one_line(&self) -> bool {
+        self.vimgrep
+    }
+
+    fn column(&self) -> bool {
+        self.vimgrep
+    }
+
+    fn contexts(&self) -> (usize, usize) {
+        let both = self.context.unwrap_or(0);
+        if both > 0 {
+            (both, both)
```

Comment:
This mimics `ripgrep`'s logic but per PR comment, if you currently supplied a combination of `--context` and either `--after-context`/`--before-context`, since we don't do the "last one wins" command-line-option-handling logic, that means that you'd get the maybe unintuitive behavior that `--context` "wins"/"outweighs" the more specific additionally-provided `--after-context`/`--before-context` value

But I'm not too concerned with that at the moment, it's "weird"/"non-optimal" usage